crypto/ring-signature/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! MobileCoin transaction data types, transaction construction and validation
+//! routines
+
+#![no_std]
+#![deny(missing_docs)]
+
+extern crate alloc;
+
+use crate::onetime_keys::create_shared_secret;
+use mc_crypto_keys::{RistrettoPrivate, RistrettoPublic};
+
+mod amount;
+mod domain_separators;
+mod token;
```

Comment:
In the past James spoke against making submodules within crates pub, because usually, the reason you make these modules is to organize the code logically within the crate to ease maintenance. But you don't need to expose the organizational details to the consumer of your crate, and doing so may break downstream if you try to reorganize later.

(paging @jcape , let's see if he agrees)

Reply:
Yeah, my preference is to expose everything at the crate's top-level export, and not force downstream to know about the modules layout in my code.